pkg/kubernetes-mcp-server/cmd/root.go

```diff
@@ -1,140 +1,180 @@
-			klog.V(0).Infof("SSE server starting on port %d and path /sse", ssePort)
-			if err := sseServer.Start(fmt.Sprintf(":%d", ssePort)); err != nil {
-				klog.Errorf("Failed to start SSE server: %s", err)
-				return
-			}
-		}
+	genericiooptions.IOStreams
+}
+
+func NewMCPServerOptions(streams genericiooptions.IOStreams) *MCPServerOptions {
+	return &MCPServerOptions{
+		IOStreams:  streams,
+		Profile:    "full",
+		ListOutput: "table",
+	}
+}
 
-		httpPort := viper.GetInt("http-port")
-		if httpPort > 0 {
-			httpServer := mcpServer.ServeHTTP()
-			klog.V(0).Infof("Streaming HTTP server starting on port %d and path /mcp", httpPort)
-			if err := httpServer.Start(fmt.Sprintf(":%d", httpPort)); err != nil {
-				klog.Errorf("Failed to start streaming HTTP server: %s", err)
-				return
+func NewMCPServer(streams genericiooptions.IOStreams) *cobra.Command {
+	o := NewMCPServerOptions(streams)
+	cmd := &cobra.Command{
+		Use:     "kubernetes-mcp-server [command] [options]",
+		Short:   "Kubernetes Model Context Protocol (MCP) server",
+		Long:    long,
+		Example: examples,
+		RunE: func(c *cobra.Command, args []string) error {
+			if err := o.Complete(); err != nil {
+				return err
+			}
+			if err := o.Validate(); err != nil {
+				return err
+			}
+			if err := o.Run(); err != nil {
+				return err
 			}
-		}
 
-		if err := mcpServer.ServeStdio(); err != nil && !errors.Is(err, context.Canceled) {
-			panic(err)
-		}
-	},
+			return nil
+		},
+	}
+
+	cmd.Flags().BoolVar(&o.Version, "version", o.Version, "Print version information and quit")
+	cmd.Flags().IntVar(&o.LogLevel, "log-level", o.LogLevel, "Set the log level (from 0 to 9)")
+	cmd.Flags().IntVar(&o.SSEPort, "sse-port", o.SSEPort, "Start a SSE server on the specified port")
+	cmd.Flags().IntVar(&o.HttpPort, "http-port", o.HttpPort, "Start a streamable HTTP server on the specified port")
+	cmd.Flags().StringVar(&o.SSEBaseUrl, "sse-base-url", o.SSEBaseUrl, "SSE public base URL to use when sending the endpoint message (e.g. https://example.com)")
+	cmd.Flags().StringVar(&o.Kubeconfig, "kubeconfig", o.Kubeconfig, "Path to the kubeconfig file to use for authentication")
+	cmd.Flags().StringVar(&o.Profile, "profile", o.Profile, "MCP profile to use (one of: "+strings.Join(mcp.ProfileNames, ", ")+")")
+	cmd.Flags().StringVar(&o.ListOutput, "list-output", o.ListOutput, "Output format for resource list operations (one of: "+strings.Join(output.Names, ", ")+")")
+	cmd.Flags().BoolVar(&o.ReadOnly, "read-only", o.ReadOnly, "If true, only tools annotated with readOnlyHint=true are exposed")
+	cmd.Flags().BoolVar(&o.DisableDestructive, "disable-destructive", o.DisableDestructive, "If true, tools annotated with destructiveHint=true are disabled")
+
+	return cmd
 }
 
-func Execute() {
-	if err := rootCmd.Execute(); err != nil {
-		klog.Errorf("Failed to execute command: %s", err)
-		os.Exit(1)
-	}
+func (m *MCPServerOptions) Complete() error {
+	m.initializeLogging()
+
+	return nil
```

Comment:
What's the purpose of this method?

Is it to match [this plugin structure](https://github.com/kubernetes/sample-cli-plugin/blob/4398d1b3b9e13a934d92ee6bb691e8af8f97b806/pkg/cmd/ns.go#L110)?

Reply:
That said; these https://github.com/manusa/kubernetes-mcp-server/blob/2c18ca0822b51bf7de8d41baada87475cd446525/pkg/kubernetes-mcp-server/cmd/root.go#L126-L138 can be moved to Complete and Validate

This https://github.com/manusa/kubernetes-mcp-server/blob/2c18ca0822b51bf7de8d41baada87475cd446525/pkg/kubernetes-mcp-server/cmd/root.go#L140-L143 should be a separate command `kubernete-mcp-server version` instead of `--version`

and There should be 3 managers -- stdio, sse, the other and `Run` should execute the requested one